### ISSUE TITLE
Reflection support for ToLiteral

### ISSUE THREAD

**Issue by lukemorbin** [dotnetrdf/dotnetrdf#212]
Hi,

I had a requirement to reflect over a class and generate literal values for each member of the class, I couldn't find anything in the documentation regarding reflection and the ```ToLiteral()``` method.

I created an additional extension method that accepts an object, finds its type at run time, switches on type, and calls the ```ToLiteral()``` method for the concrete type. You can create a properly namespaced literal value by calling ```ReflectToLiteral(object obj)```.

The name is slightly crass however when it was called simply ```ToLiteral(object obj)``` it would loop ad infinitum. 

This approach seems to work for my implementation and thought I'd bring it to your attention in case it was useful.

You can see the extension class here;
https://github.com/lukemorbin/strava-rdf-cs/blob/f3730088cba544ca84f7b8568c4fdf3d6b7bdfdb/LiteralExtensions.cs

**Comment by tpluscode** [comment]
Does it handle all the same [LiteralExpressionExtensions](https://github.com/tpluscode/dotnetrdf/blob/76b1c3d672b8823ced5dd04c86772b35135c48e7/Libraries/dotNetRDF/Query/Builder/Expressions/LiteralExpressionExtensions.cs)?

Guess we could make it open and put outs query builder namespace.

By the way, note the type matching `switch` supported by C# 7.1 :)

**Comment by lukemorbin** [comment]
Yeah it handles the same literal expressions as yours you linked.

The one you linked is essentially the same thing (done in a better way) by the looks of it. 
Could be useful to make it useable by library users?
